src/room_buffer.rs

```diff
@@ -196,11 +204,89 @@ impl RoomBuffer {
         }
     }
 
-    pub fn update_buffer_name(&mut self) {
+    pub fn update_buffer_name(&self) {
         let name = self.calculate_buffer_name();
         self.weechat_buffer().set_name(&name)
     }
 
+    /// Send out a typing notice.
+    ///
+    /// This will send out a typing notice or reset one if needed. It will make
+    /// sure that only one typing notice request is in flight at a time.
```

Comment:
```suggestion
    /// This will send out a typing notice or reset the one in progress, if
    /// needed. It will make sure that only one typing notice request is in
    /// flight at a time.
```